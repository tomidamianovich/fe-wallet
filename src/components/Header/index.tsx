import React from 'react'
import './styles/index.scss'
import logo from '../../assets/logo-brand.svg'
import Icon from '../Icon'
import { NavLink } from 'react-router-dom'

/*
  Component Name: Header
  Usages: 
    - Component header that allows a logo and various features in the right side of the same one.
    
*/

const Header: React.FC = () => (
  <header className="fe-wallet__header">
    <div className="fe-wallet__header__logo-container">
      <NavLink to="/fe-wallet">
        <img
          src={logo}
          alt={'logo'}
          className="fe-wallet__header__logo-container__logo"
        />
      </NavLink>
      <div className="fe-wallet__header__icons">
        <Icon icon="help" width={26} height={26} />
        <Icon icon="bell" width={32} height={32} />
        <Icon icon="profile" width={32} height={29} />
      </div>
    </div>
  </header>
)
export default Header
