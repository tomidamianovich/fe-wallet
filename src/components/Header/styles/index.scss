@import "../../../containers/App/styles/colours.scss";

.fe-wallet {

  &__header {
    background-color: $primary-background-color;
    height: 80px;
    display: flex;
    margin: 0;
    padding: 0;

    &__logo-container {
      height: 80px;
      width: 68px;
      display: flex;
      left: 0px;
      top: 0px;
      border-radius: 0px;
      align-items: center;
      justify-content: flex-start;
      padding-left: 32px;

      &__logo {
        color: $primary-base-color;
        height: 32.8px;
        width: 68px;
      }
    }
  }
}