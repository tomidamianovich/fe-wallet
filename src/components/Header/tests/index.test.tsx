import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Header from '../'

test('renders Exchange main page', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>,
  )
  expect(asFragment()).toMatchSnapshot()
})
