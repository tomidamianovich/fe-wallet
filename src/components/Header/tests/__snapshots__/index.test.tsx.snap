// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Exchange main page 1`] = `
<DocumentFragment>
  <header
    class="fe-wallet__header"
  >
    <div
      class="fe-wallet__header__logo-container"
    >
      <a
        class=""
        href="/panel"
      >
        <img
          class="fe-wallet__header__logo-container__logo"
          src="logo-brand.svg"
        />
      </a>
    </div>
  </header>
</DocumentFragment>
`;
